src/service/resources-impl.js

```diff
@@ -1037,11 +1046,23 @@ export class Resources {
    * @private
    */
   calcTaskTimeout_(task) {
+    const now = timer.now();
+
     if (this.exec_.getSize() == 0) {
-      return 0;
+      // If we've never been visible, return 0. This follows the previous
+      // behavior of not delaying tasks when there's nothing to do.
+      // TODO(jridgewell): Should we return an un-subtracted `penalty` here?
```

Comment:
I don't think TODO is needed here. The processing is done via two queues: scheduling and execution. We delay execution here which is fine. If we have an issue with INVISIBLE mode - that should be addressed in scheduling and it already is - i.e. we do not schedule elements unless they opt-in and close to viewport, etc.